internal/models/config.go

```diff
@@ -0,0 +1,171 @@
+type PipelineReceiver struct {
+	Name string `mapstructure:"name"`
+}
+
+// PipelineOperation defines an operation of a pipeline.
+type PipelineOperation struct {
+	Name string `mapstructure:"name"`
+}
+
+// PipelineExporter defines an exporter of a pipeline.
+type PipelineExporter struct {
+	Name string `mapstructure:"name"`
+}
+
+// NamedEntity defines a configuration entity that has a name.
+type NamedEntity interface {
+	GetName() string
+}
+
+// ReceiverCfg is the configuration of a receiver. Specific receivers
+// must implement this interface and will typically embed ReceiverCommon
+// struct or a struct that extends it.
+type ReceiverCfg interface {
+	NamedEntity
+}
+
+// TypeKeyName must match Type field key names for Receiver and Exporter.
+const TypeKeyName = "type"
+
+// ExporterCfg is the configuration of an exporter. Specific exporters
+// must implement this interface and will typically embed ExporterCommon
+// struct or a struct that extends it.
+type ExporterCfg interface {
+	NamedEntity
+}
+
+// OptionCfg is the configuration of a processing option. Specific options
+// must implement this interface and will typically embed OptionCommon
+// struct or a struct that extends it.
+type OptionCfg interface {
+	NamedEntity
+}
+
+// Below are common setting structs for Receivers, Exporters and Options.
+// These are helper structs which you can use to implement your specific
+// receiver/exporter/option config storage.
+
+// ReceiverCommon defines common settings for a single-protocol receiver configuration.
+// Specific receivers can embed this struct and extend it with more fields if needed.
+type ReceiverCommon struct {
+	Type    string `mapstructure:"type"`
+	Enabled bool   `mapstructure:"enabled"`
+	Name    string `mapstructure:"name"`
+	Address string `mapstructure:"address"`
+	Port    int    `mapstructure:"port"`
+}
+
+// GetName returns receiver name.
+func (cfg *ReceiverCommon) GetName() string {
+	return cfg.Name
+}
+
+// ExporterCommon defines common settings for an exporter configuration.
+// Specific exporters can embed this struct and extend it with more fields if needed.
+type ExporterCommon struct {
+	Type    string `mapstructure:"type"`
+	Enabled bool   `mapstructure:"enabled"`
+	Name    string `mapstructure:"name"`
+}
+
+// MultiReceiverCommon defines common settings for a multi-protocol receiver configuration.
+// Specific receivers can embed this struct and extend it with more fields. Note that
+// Protocols field is not defined, you need to define it with a specific type.
+type MultiReceiverCommon struct {
+	Type string `mapstructure:"type"`
+	Name string `mapstructure:"name"`
+}
+
+// GetName returns receiver name.
+func (cfg *MultiReceiverCommon) GetName() string {
+	return cfg.Name
+}
+
+// ProtocolReceiver defines a protocol of multi-protocol receiver.
+// Specific receivers can embed this struct and extend it with more fields if needed.
+type ProtocolReceiver struct {
```

Comment:
Not sure why this is needed. Why 80% of the fields are duplicated with ReceiverCommon?

Reply:
You are right, it is not clear what for this is because it is not used in this PR. It is used in a subsequent PR. I will remove it from this one. It will be clearer in the next one.